sci-libs/vtk/vtk-9.1.0.ebuild

```diff
@@ -0,0 +1,601 @@
+	mpi mysql odbc offscreen openmp postgres python qt5 +rendering tbb test
+	theora +threads tk video_cards_nvidia views web"
+
+RESTRICT="!test? ( test )"
+
+REQUIRED_USE="
+	all-modules? ( boost ffmpeg gdal imaging mysql odbc postgres qt5 rendering theora views )
+	cuda? ( video_cards_nvidia )
+	java? ( rendering )
+	python? ( ${PYTHON_REQUIRED_USE} )
+	qt5? ( rendering )
+	tk? ( rendering python )
+	web? ( python )
+"
+
+RDEPEND="
+	app-arch/lz4:=
+	app-arch/xz-utils
+	dev-db/sqlite:3
+	dev-cpp/eigen[cuda?,openmp?]
+	dev-libs/double-conversion:=
+	dev-libs/expat
+	dev-libs/icu:=
+	dev-libs/jsoncpp:=
+	dev-libs/libxml2:2
+	dev-libs/pugixml
+	media-libs/freetype
+	media-libs/libogg
+	media-libs/libpng:=
+	media-libs/libtheora
+	media-libs/tiff
+	<sci-libs/hdf5-1.12:=[mpi=]
+	sci-libs/netcdf:=[mpi=]
+	sys-libs/zlib
+	virtual/jpeg
+	all-modules? ( sci-geosciences/liblas[gdal] )
+	boost? ( dev-libs/boost:=[mpi?] )
+	cuda? ( dev-util/nvidia-cuda-toolkit:= )
+	ffmpeg? ( media-video/ffmpeg:= )
+	gdal? ( sci-libs/gdal:= )
+	java? ( >=virtual/jdk-1.8:* )
+	mpi? ( sys-cluster/openmpi[cxx,romio] )
+	mysql? ( dev-db/mariadb-connector-c )
+	odbc? ( dev-db/unixODBC )
+	offscreen? ( media-libs/mesa[osmesa] )
+	postgres? ( dev-db/postgresql:= )
+	python? ( ${PYTHON_DEPS} )
+	qt5? (
+		dev-qt/qtcore:5
+		dev-qt/qtsql:5
+		dev-qt/qtwidgets:5
+	)
+	rendering? (
+		media-libs/freeglut
+		media-libs/glew:=
+		media-libs/libsdl2
+		sci-libs/proj:=
+		virtual/opengl
+		x11-libs/gl2ps
+		x11-libs/libXcursor
+	)
+	tbb? ( dev-cpp/tbb:= )
+	tk? ( dev-lang/tk:= )
+	video_cards_nvidia? ( x11-drivers/nvidia-drivers[tools,static-libs] )
+	views? (
+		x11-libs/libICE
+		x11-libs/libXext
+	)
+	web? ( ${WEBAPP_DEPEND} )
+	$(python_gen_cond_dep '
+		python? (
+			boost? ( dev-libs/boost:=[mpi?,python?,${PYTHON_USEDEP}] )
+			gdal? ( sci-libs/gdal:=[python?,${PYTHON_USEDEP}] )
+			mpi? ( dev-python/mpi4py[${PYTHON_USEDEP}] )
+		)
+	')
+"
+DEPEND="
+	${RDEPEND}
+	<dev-libs/pegtl-3
+	dev-libs/utfcpp
+"
+BDEPEND="
+	virtual/pkgconfig
+	mpi? ( app-admin/chrpath )
+"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-9.0.1-0001-fix-kepler-compute_arch-if-CUDA-toolkit-11-is-used.patch
+	"${FILESDIR}"/${PN}-8.2.0-freetype-2.10.3-provide-FT_CALLBACK_DEF.patch
+)
+
+DOCS=( CONTRIBUTING.md README.md )
+
+CHECKREQS_DISK_BUILD="3G"
+# we want the EAPI 8 default
+CMAKE_BUILD_TYPE=RelWithDebInfo
+
+pkg_pretend() {
+	[[ MERGE_TYPE != binary ]] && has openmp && tc-check-openmp
+
+	if use examples; then
+		CHECKREQS_DISK_BUILD="4G"
+	fi
+	if use cuda; then
+		# NOTE: This should actually equal to (number of build jobs)*7G,
+		# as any of the cuda compile tasks can take up 7G!
+		# 10.2 GiB install directory, 6.4 GiB build directory with max. USE flags
+		CHECKREQS_MEMORY="7G"
+		CHECKREQS_DISK_BUILD="14G"
+	fi
+	check-reqs_pkg_setup
+}
+
+pkg_setup() {
+	[[ MERGE_TYPE != binary ]] && has openmp && tc-check-openmp
+
+	if use examples; then
+		CHECKREQS_DISK_BUILD="4G"
+	fi
+	if use cuda; then
```

Comment:
nit: new line between ifs

Reply:
The use examples if-clause will probably be removed anyway. With default USE flags, the package already needs more than 3G, so increased the default value to 4G in my current working copy.